### ISSUE TITLE
[Bug] error on startup and widgets not loading

### ISSUE THREAD

**Issue by Blahkaey** [drinfernoo/plugin.program.autowidget#129]
On startup I get an error pop up for auto widget and widgets never load, they just say loading content. Shortcuts do seem to work though.

I’m on Xbox series x kodi 19.4 using titan binge and venom

- NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
Error Type: <class 'AssertionError'>
Traceback (most recent call last):
File "/addons\plugin.program.autowidget\service.py", line 4, in <module>
monitor = refresh.RefreshService(
File "/addons\plugin.program.autowidget\resources\lib\refresh.py", line 33, in _init,
self. update widgets(
File "/addons\plugin.program.autowidget\resources\lib\refresh.py",line 121, in _update_widgets
cache.cache and updatel
File "/addons\plugin.program.autowidget\resources\lib\common\cache.py",line 152, in cache_and_update
assert widget_id
AssertionError

**Comment by dmyoung9** [comment]
I'll need a full log if anything 🤔

**Comment by drinfernoo** [comment]
I'll need a full log if anything 🤔

**Comment by fuxkuptheworld** [comment]
I'm only receiving the pop up error message on one skin which is titan mod but the error is still in the logs on every kodi boot

log files
https://paste.kodi.tv/eheguduzif.kodi


![Kodi_20220420_165542](https://user-images.githubusercontent.com/104105124/164327415-e3792d4f-eb29-45f2-9235-ab08b56c0c31.png)
![Kodi_20220420_165551](https://user-images.githubusercontent.com/104105124/164327428-1f514fa7-1dd7-4134-94ab-7a23cedc8c74.png)

**Comment by Blahkaey** [comment]
https://paste.kodi.tv/puculocozu

**Comment by drinfernoo** [comment]
You know, I'm not really sure why this would be happening 🤔 The assertion that's erroring is based on a widget ID that should come from one of the cache files, which shouldn't ever be malformed, but it could happen. Can you try clearing your cache from AutoWidget's Settings or `Tools -> Clear Cache`?

**Comment by Hawkedon** [comment]
Hi @drinfernoo, I got this issue too, seems it's due to the Xbox access restriction for some os python package functionality.

@blakie1225

Edit: After debugging the addon with Web-PDB, I finally figured out the problem for Xbox. Just removing the three assertions at the beginning of cache_and_update method will fix the crash. The widget_id variable is None due to the yield statement. The assertion test will fail for the variable, but on PC (what I tested), the execution of it will be ignored.
